src/app/codegen-interaction-model/Model.cpp

```diff
@@ -0,0 +1,267 @@
+}
+
+} // namespace
+
+CHIP_ERROR Model::ReadAttribute(const InteractionModel::ReadAttributeRequest & request, InteractionModel::ReadState & state,
+                                AttributeValueEncoder & encoder)
+{
+    // TODO: this needs an implementation
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+CHIP_ERROR Model::WriteAttribute(const InteractionModel::WriteAttributeRequest & request, AttributeValueDecoder & decoder)
+{
+    // TODO: this needs an implementation
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+CHIP_ERROR Model::Invoke(const InteractionModel::InvokeRequest & request, chip::TLV::TLVReader & input_arguments,
+                         InteractionModel::InvokeReply & reply)
+{
+    // TODO: this needs an implementation
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+EndpointId Model::FirstEndpoint()
+{
+    // find the first enabled index
+    const uint16_t lastEndpointIndex = emberAfEndpointCount();
+    for (uint16_t endpointIndex = 0; endpointIndex < lastEndpointIndex; endpointIndex++)
+    {
+        if (emberAfEndpointIndexIsEnabled(endpointIndex))
+        {
+            return emberAfEndpointFromIndex(endpointIndex);
+        }
+    }
+
+    // No enabled endpoint found. Give up
+    return kInvalidEndpointId;
+}
+
+EndpointId Model::NextEndpoint(EndpointId before)
+{
+    // find the first enabled index
+    bool beforeFound = false;
+
+    const uint16_t lastEndpointIndex = emberAfEndpointCount();
+    for (uint16_t endpointIndex = 0; endpointIndex < lastEndpointIndex; endpointIndex++)
+    {
+        if (!beforeFound)
+        {
+            beforeFound = (before == emberAfEndpointFromIndex(endpointIndex));
+            continue;
+        }
+
+        if (emberAfEndpointIndexIsEnabled(endpointIndex))
+        {
+            return emberAfEndpointFromIndex(endpointIndex);
+        }
+    }
+
+    // No enabled enpoint after "before" was found, give up
+    return kInvalidEndpointId;
+}
+
+InteractionModel::ClusterEntry Model::FirstCluster(EndpointId endpointId)
+{
+    const EmberAfEndpointType * endpoint = emberAfFindEndpointType(endpointId);
+    VerifyOrReturnValue(endpoint != nullptr, InteractionModel::ClusterEntry::Invalid());
+    VerifyOrReturnValue(endpoint->clusterCount > 0, InteractionModel::ClusterEntry::Invalid());
+    VerifyOrReturnValue(endpoint->cluster != nullptr, InteractionModel::ClusterEntry::Invalid());
+
+    return FirstServerClusterEntry(endpointId, endpoint, 0);
+}
+
+InteractionModel::ClusterEntry Model::NextCluster(const ConcreteClusterPath & before)
+{
+    const EmberAfEndpointType * endpoint = emberAfFindEndpointType(before.mEndpointId);
+    VerifyOrReturnValue(endpoint != nullptr, InteractionModel::ClusterEntry::Invalid());
+    VerifyOrReturnValue(endpoint->clusterCount > 0, InteractionModel::ClusterEntry::Invalid());
+    VerifyOrReturnValue(endpoint->cluster != nullptr, InteractionModel::ClusterEntry::Invalid());
+
+    for (uint16_t i = 0; i < endpoint->clusterCount; i++)
```

Comment:
suggest all the `i` be renamed like `cluster_idx` to make it clear what value it is (e.g. `endpoint_idx`, `cluster_idx`).